app/models/concerns/operations/stock_operations.rb

```diff
@@ -62,11 +62,25 @@ def register_loss(package, quantity:, from_location:)
         end
       end
 
-      PackagesInventory.append_loss(
+      PackagesInventory.public_send("append_#{action}", {
         package: package,
         quantity: quantity.abs * -1,
-        location: from_location
-      )
+        location_id: location_id,
+        description: description
+      })
+    end
+
+    def perform_action(package, quantity:, location_id:, action: 'loss', description: nil)
+      if PackagesInventory::QUANTITY_LOSS_ACTIONS.include?(action)
+        register_loss(
+          package,
+          quantity: quantity,
+          location_id: location_id,
+          action: action,
+          description: description)
+      else
+        raise Goodcity::ActionNotAllowedError.new
```

Comment:
Are we not allowing gains yet ? Don't think there's anything preventing us from doing so.

Also, what do you think of changing the name of the method to `register_quantity_change` , to be more specific on what it does

Reply:
For this ticket, we are only considering actions related to `loss`. The `gain` related actions and undo actions will be covered in another ticket.

so once we have gain related actions, will add `PackagesInventory::QUANTITY_GAIN_ACTIONS` and if falls under that then will call `register_gain` for it.

> Also, what do you think of changing the name of the method to register_quantity_change , to be more specific on what it does

yes, will rename it from `perform_action` to `register_quantity_change`.

Reply:
@swatijadhav Sounds good, was just wondering :)